fdbclient/include/fdbclient/GetEncryptCipherKeys.actor.h

```diff
@@ -329,5 +329,21 @@ Future<TextAndHeaderCipherKeys> getEncryptCipherKeys(Reference<AsyncVar<T> const
 	return result;
 }
 
+ACTOR template <class T>
+Future<TextAndHeaderCipherKeys> getEncryptCipherKeys(Reference<AsyncVar<T> const> db,
```

Comment:
With configurable encryption its possible for the header to be empty (invalid encrypt domain for `ENCRYPT_HEADER_AUTH_TOKEN_MODE_NONE`). In that case we should not request for header cipher keys at it will return `key_not_found` errors.

Reply:
The struct I plan to expose when `CipherDetails` is asked from the header is planned to be the following:

```
struct EncryptHeaderCipherDetails {
	Optional<BlobCipherDetails> textCipherDetails;
	Optional<BlobCipherDetails> headerCipherDetails;
};
```

So yes, for authTokenMode == None, there won't be any headerCipherKey or details

Reply:
I'm curious when would it be possible for textCipherDetails to not be present?

Reply:
When encryption is disabled :D

Reply:
:) I see, i would assume we won't use these data structures when encryption is disabled so maybe we can not make that field optional. Otherwise it might be confusing